### PR TITLE
Extend to include CoordFlip

### PR DESCRIPTION
**Pull request by ijlyttle**
@wenyuyangpku - I will let you know when this is ready for ggvega

### CODE REVIEW

**Comment by ijlyttle** [comment]
I have put an empty function into ggvega for CoordFlip, just so it would build.

I will write up the development-document next.

**Comment by ijlyttle** [comment]
@wenyuyangpku - I think this is ready for you now.

There are two files you should be aware of:

- the [development document](https://github.com/vegawidget/ggvega/blob/coord-flip/dev/exploratory/coord-flip.md), showing the what the vegaspec should look like.

- the [deployment document](https://github.com/vegawidget/ggvega/blob/coord-flip/data-raw/examples/deploy.Rmd) - when you think you are ready to test, remove `"scat-coord-flip-iris"` from `skip_vegaspec`, then re-render the document (hit the "knit" button in the RStudio IDE): 

https://github.com/vegawidget/ggvega/blob/22a0a11af08dcfcfc44e226854f8bcf05b19ca97/data-raw/examples/deploy.Rmd#L89-L91

**Comment by wenyuyangpku** [comment]
@ijlyttle - I have done.

Our framework looks good.

File: src-ext/TypeScript/ggvega/src/itmLayerArrayByCoord.ts [hunk 1]
```diff
@@ -70,5 +70,13 @@ function itmLayerArrayByCoordCartesian(itmLayerArray: ItmLayer[], gsCoord: GG.Co
  */
 function itmLayerArrayByCoordFlip(itmLayerArray: ItmLayer[], gsCoord: GG.Coord): ItmLayer[] {
   // will do something
+
+  // exchange encoding.x and encoding.y
```
Comment (ijlyttle) [inline src-ext/TypeScript/ggvega/src/itmLayerArrayByCoord.ts]: This has to be more robust than looking at just `x` and `y`. It needs to take into account other aesthetics like `xmin`, `ymin`.

Comment (wenyuyangpku) [reply src-ext/TypeScript/ggvega/src/itmLayerArrayByCoord.ts]: Sorry, I have no idea. Can you give me a specific example?

As my understanding, the `Vl.Encoding` only has `x` and `y`. And `xmin`, `ymin` is part of them(for now, they could be part of the `scale`. Maybe there are other situations). So exchange `x` and `y` is enough?

**Pull request merged**